core/src/main/java/bio/terra/pearl/core/service/survey/SurveyService.java

```diff
@@ -94,6 +102,27 @@ public Survey create(Survey survey) {
         return savedSurvey;
     }
 
+    private List<ReferencedQuestion> parseAndCreateReferencedQuestions(Survey survey) {
+        // loop through potential referenced questions - if we can find the question in the db, create a ReferencedQuestion
+        SurveyParseUtils.findPotentialReferencedSurveyQuestions(survey).forEach((potentialSurveyStableId, potentialQuestionIds) -> {
+            potentialQuestionIds.forEach(potentialQuestionId -> {
+                Optional<SurveyQuestionDefinition> questionDef = surveyQuestionDefinitionDao.findByStableId(potentialSurveyStableId, potentialQuestionId);
+
+                if (questionDef.isPresent()) {
+                    ReferencedQuestion referencedQuestion = new ReferencedQuestion();
+                    referencedQuestion.setSurveyId(survey.getId());
+
+                    referencedQuestion.setReferencedSurveyStableId(potentialSurveyStableId);
+                    referencedQuestion.setReferencedQuestionStableId(potentialQuestionId);
+
+                    referencedQuestionDao.create(referencedQuestion);
+                }
+            });
```

Comment:
I'm debating if this check is worth it.

On the one hand, it's nice to be able to confirm that we're not loading junk data every time someone opens a survey - for example, if someone uses a matrix question `{myMatrixQuestion.myAnswer}`, we don't want a referenced answer to be created for that. Without this check, it would be created, with the survey stable id of `myMatrixQuestion`.

On the other hand, this introduces the possibility of a race condition on populate and general confusion. It could be that we just need to make good UI around this so that if you reference a question it's clear in the UI if the reference was picked up by the system or not.